src/sqlite3gen.cpp

```diff
@@ -138,55 +206,71 @@ const char * schema_queries[][2] = {
       "\taddable              INTEGER DEFAULT 0, -- 0:no 1:yes\n"
       "\tremovable            INTEGER DEFAULT 0, -- 0:no 1:yes\n"
       "\traisable             INTEGER DEFAULT 0, -- 0:no 1:yes\n"
-      /// @todo make a `kind' table
-      "\tkind                 INTEGER DEFAULT 0, -- 0:define 1:function 2:variable 3:typedef 4:enum 5:enumvalue 6:signal 7:slot 8:friend 9:DCOP 10:property 11:event\n"
+      "\tkind                 TEXT NOT NULL, -- 'macro definition' 'function' 'variable' 'typedef' 'enumeration' 'enumvalue' 'signal' 'slot' 'friend' 'dcop' 'property' 'event' 'interface' 'service'\n"
       "\tbodystart            INTEGER DEFAULT 0, -- starting line of definition\n"
       "\tbodyend              INTEGER DEFAULT 0, -- ending line of definition\n"
-      "\tid_bodyfile          INTEGER DEFAULT 0, -- file of definition\n"
-      "\tid_file              INTEGER NOT NULL,  -- file where this identifier is located\n"
+      "\tbodyfile_id          INTEGER REFERENCES path, -- file of definition\n"
+      "\tfile_id              INTEGER NOT NULL REFERENCES path,  -- file where this identifier is located\n"
       "\tline                 INTEGER NOT NULL,  -- line where this identifier is located\n"
       "\tcolumn               INTEGER NOT NULL,  -- column where this identifier is located\n"
-      /// @todo make a `detaileddescription' table
       "\tdetaileddescription  TEXT,\n"
       "\tbriefdescription     TEXT,\n"
-      "\tinbodydescription    TEXT\n"
-      ");"
+      "\tinbodydescription    TEXT,\n"
+      "\tFOREIGN KEY (rowid) REFERENCES refid (rowid)\n"
+    ");"
+  },
+  { "member",
+    "CREATE TABLE IF NOT EXISTS member (\n"
+      "\t-- Memberdef <-> containing compound relation.\n"
+      "\t-- Similar to XML listofallmembers.\n"
+      "\trowid            INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,\n"
+      "\tscope_rowid      INTEGER NOT NULL REFERENCES compounddef,\n"
+      "\tmemberdef_rowid  INTEGER NOT NULL REFERENCES memberdef,\n"
+      "\tprot             INTEGER NOT NULL,\n"
+      "\tvirt             INTEGER NOT NULL,\n"
+      "\tUNIQUE(scope_rowid, memberdef_rowid)\n"
```

Comment:
The XML `<member>` node can have inner `<scope>` and `<name>` nodes. I couldn't find any `<scope>` that was anything but the current containing scope, nor a `<name>` that wasn't just the name of the memberdef at the corresponding refid.

I've left these out of SQL because they smell like they're just convenience features in the XML. I'll need to rethink this If they *can* actually differ.